mirrord/cli/src/user_data.rs

```diff
@@ -0,0 +1,141 @@
+use std::{path::PathBuf, sync::LazyLock};
+
+use fs4::tokio::AsyncFileExt;
+use serde::{Deserialize, Serialize};
+use tokio::{
+    fs,
+    io::{self, AsyncReadExt, AsyncWriteExt},
+};
+use tracing::trace;
+use uuid::Uuid;
+
+/// "~/.mirrord"
+static DATA_STORE_DIR: LazyLock<PathBuf> = LazyLock::new(|| {
+    home::home_dir()
+        .unwrap_or_else(|| PathBuf::from("~"))
+        .join(".mirrord")
+});
+
+/// "~/.mirrord/data.json"
+static DATA_STORE_PATH: LazyLock<PathBuf> = LazyLock::new(|| DATA_STORE_DIR.join("data.json"));
+
+/// Data that we store in the user's machine at `~/.mirrord/data.json` that might be used
+/// for a variety of purposes.
+///
+/// Whenever we deserialize the `UserData` json file, if there are any errors we generate
+/// a new one using [`UserData::default`]. To avoid overwriting the file with all new default
+/// values in case we got a deserialization error due to a missing field, each field here
+/// gets a `default` annotation, so only the missing fields will be updated.
+#[derive(Debug, Serialize, Deserialize)]
+pub(crate) struct UserData {
+    /// Amount of times this user has run mirrord.
+    #[serde(default)]
+    session_count: u32,
+
+    /// Helps us keep track of unique users for analytics when telemetry is enabled.
+    ///
+    /// Must use custom `default =`, since the default is [`Uuid::nil`].
+    #[serde(default = "default_uuid")]
```

Comment:
```suggestion
    #[serde(default = "Uuid::new_v4")]
```
This should work, I think